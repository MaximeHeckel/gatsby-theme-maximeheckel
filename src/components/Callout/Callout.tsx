import React from 'react';
import styled, { Theme } from '../../utils/styled';

export enum VARIANT {
  DANGER = 'danger',
  INFO = 'info',
}

interface Props {
  variant: VARIANT;
}

const Callout: React.FC<Props> = (props) => (
  <StyledCallout {...props}>{props.children}</StyledCallout>
);

const variantColors = (theme: Theme): Record<string, string> => ({
  [VARIANT.DANGER]: `
    background-color: #FFEFEE;
    border-color: ${theme.colors.red};
    color: #2B2D3E;
    `,
  [VARIANT.INFO]: `
    background-color: #E6EDF8;
    border-color: ${theme.colors.blue};
    color: #2B2D3E;
  `,
});

const StyledCallout = styled('div')<{ variant: VARIANT }>`
  border-radius: 4px;
  border-left: 3px solid;
  padding: 30px 30px;
  margin-bottom: 25px;

  *:last-child {
    margin-bottom: 0px;
  }

  ${(p) => variantColors(p.theme)[p.variant]}
`;

export { Callout };
