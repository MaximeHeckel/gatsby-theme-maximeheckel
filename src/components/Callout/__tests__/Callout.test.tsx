import { cleanup, render } from '@testing-library/react';
import React from 'react';
import Callout from '../';
import { VARIANT } from '../Callout';

describe('Callout', () => {
  afterEach(cleanup);
  it('can render a Callout with variant info', () => {
    const component = render(<Callout variant={VARIANT.INFO}>Test</Callout>);
    expect(component.baseElement).toMatchSnapshot();
  });

  it('can render a Callout with variant danger', () => {
    const component = render(<Callout variant={VARIANT.DANGER}>Test</Callout>);
    expect(component.baseElement).toMatchSnapshot();
  });
});
