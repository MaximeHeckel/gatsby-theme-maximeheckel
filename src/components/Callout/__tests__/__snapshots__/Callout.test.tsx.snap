// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Callout can render a Callout with variant danger 1`] = `
.emotion-0 {
  border-radius: 4px;
  border-left: 3px solid;
  padding: 30px 30px;
  margin-bottom: 25px;
  background-color: #FFEFEE;
  border-color: #FF515A;
  color: #2B2D3E;
}

.emotion-0 *:last-child {
  margin-bottom: 0px;
}

<body>
  <div>
    <div
      class="emotion-0"
    >
      Test
    </div>
  </div>
</body>
`;

exports[`Callout can render a Callout with variant info 1`] = `
.emotion-0 {
  border-radius: 4px;
  border-left: 3px solid;
  padding: 30px 30px;
  margin-bottom: 25px;
  background-color: #E6EDF8;
  border-color: var(--maximeheckel-colors-brand,#5184f9);
  color: #2B2D3E;
}

.emotion-0 *:last-child {
  margin-bottom: 0px;
}

<body>
  <div>
    <div
      class="emotion-0"
    >
      Test
    </div>
  </div>
</body>
`;
